lambda/matching.ml

```diff
@@ -97,14 +97,8 @@ let lforget {left=left ; right=right} = match right with
 | _::xs -> {left=omega::left ; right=xs}
 |  _ -> assert false
 
-let rec small_enough n = function
-  | [] -> true
-  | _::rem ->
-      if n <= 0 then false
-      else small_enough (n-1) rem
-
 let ctx_lshift ctx =
-  if small_enough (!Clflags.match_context_rows - 1) ctx then
+  if List.length ctx < !Clflags.match_context_rows then
```

Comment:
Commit message: "simplify ctx_lshift".

I would be okay with `List.compare_length_with ctx !Cflags.match_context_rows < 0`, although I don't find it as readable as the current version.

Reply:
I agree it's not as readable.
It's just that in the absence of benchmarks, I am a little bit worried by the various changes we've done that have the potential to degrade compilation time.

If you are not worried, I'm happy to keep things as is for now (and perhaps do another pass dedicated to speeding things up at a later date if it appears necessary)